libs/dh/shared/util-application-insights/src/lib/dh-application-insights.service.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Inject, Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { AngularPlugin } from '@microsoft/applicationinsights-angularplugin-js';
+import {
+  ApplicationInsights,
+  IExceptionTelemetry,
+  IPageViewTelemetry,
+} from '@microsoft/applicationinsights-web';
+
+import {
+  DhAppEnvironmentConfig,
+  dhAppEnvironmentToken,
+} from '@energinet-datahub/dh/shared/environments';
+
+@Injectable({ providedIn: 'root' })
+export class DhApplicationInsights {
+  private angularPlugin = new AngularPlugin();
+  private appInsights = new ApplicationInsights({
+    config: {
+      instrumentationKey:
+        this.dhAppConfig.applicationInsights.instrumentationKey,
+      extensions: [this.angularPlugin],
+      extensionConfig: {
+        [this.angularPlugin.identifier]: {
+          router: this.router,
+        },
+      },
+    },
+  });
+
+  constructor(
+    private router: Router,
+    @Inject(dhAppEnvironmentToken)
+    private dhAppConfig: DhAppEnvironmentConfig
+  ) {}
+
+  /**
+   * Initialize Application Insights
+   */
+  init(): void {
+    if (this.appInsights.appInsights.isInitialized()) {
+      return;
+    }
+
+    this.appInsights.loadAppInsights();
+  }
+
+  /**
+   * Log a user action or other occurrence.
+   * @param name
+   */
+  trackEvent(name: string): void {
+    this.appInsights.trackEvent({ name });
+  }
+
+  /**
+   * Log a diagnostic scenario such entering or leaving a function.
+   * @param message
+   */
+  trackTrace(message: string): void {
+    this.appInsights.trackTrace({ message });
+  }
+
+  /**
+   * Logs that a page, or similar container was displayed to the user.
+   * @param pageView
+   */
+  trackPageView(pageView: IPageViewTelemetry): void {
```

Comment:
Consideration: `the interface of `IPageViewTelemetry` seems only to have optional properties. For the name, the `title` will be used by default. I guess this will not be ideal if the primary usage of this method will be to track when overlays are displayed.

Reply:
Fair point. Let's start with one param and expand if needed.